src/main/java/com/movie/store/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,80 @@
+package com.movie.store.dao.impl;
+
+import com.movie.store.dao.ShoppingCartDao;
+import com.movie.store.exceptions.DataProcessingException;
+import com.movie.store.lib.Dao;
+import com.movie.store.model.ShoppingCart;
+import com.movie.store.model.User;
+import com.movie.store.util.HibernateUtil;
+import lombok.extern.log4j.Log4j;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Log4j
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart add(ShoppingCart shoppingCart) {
+        log.info("Calling method add() in ShoppingCartDao");
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(shoppingCart);
+            transaction.commit();
+            return shoppingCart;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(
+                    "Can't insert shoppingCart entity. " + shoppingCart, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public ShoppingCart getByUser(User user) {
+        log.info("Calling method getByUser() in ShoppingCartDao");
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<ShoppingCart> query = session.createQuery(
+                    "from ShoppingCart s "
+                            + "left join fetch s.tickets t "
+                            + "left join fetch t.movieSession m "
```

Comment:
in ticket class you have @ManyToOne annotation for movieSession field which means that it will use Eager fetchtype(that's okay for relationship that end by "One") Try you use inner join for user and left join for tickets and see if you will have all the field of all nested objects